llvm/test/CodeGen/DirectX/finalize-linkage-remove-dead.ll

```diff
@@ -0,0 +1,222 @@
+; RUN: opt -S -dxil-finalize-linkage -mtriple=dxil-unknown-shadermodel6.5-compute %s | FileCheck %s
+; RUN: llc %s --filetype=asm -o - | FileCheck %s
```

Comment:
It might be good to have a similar test for libraries that would make sure all exported functions stay in the module despite not being called.

Reply:
Done!

Reply:
Shouldn't the exported functions stay in the module when it is compiled as a library, even though they are not called?

Reply:
Yes, and this test expects them to. The exported functions start at line 137 and all of them have checks that they remain defined.